### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Behnam1369**
Allow adding a new post and editing existing post 
Allow adding a comment 
Allow liking\unliking a post

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -26,4 +26,8 @@
       <% end %>
       </ul>
     </div>
+    <%= form_with model: @like,  url: "#{post.id}/like_toggle", method: :post , local: true do |form|%>
+      <a href=<%="#{post.id}/comments/new"%>>Add comment</a>
+      <%= form.submit "Like/Unlike this post" %>
+    <% end %>
```
Comment (Lameck1) [inline app/views/posts/show.html.erb:32]: - [ ] These two buttons need not be within a form when we consider structuring our page semantically. They can just be two separate buttons that lead to some path.

  Did you have any special reasons for wrapping these two in a form?
  
  That aside, currently, a user cannot create a like when the like button is clicked. Even though you've defined a likes path in the routes, there is no dedicated controller that handles the creation of a like to a post.
  
  Please add the `LikesController` and use it to enable liking a post.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
### Ignore this 🔴 .

**Review by Lameck1 — APPROVED** [review]
Hi, @Behnam1369,

Tasks for this project have been successfully implemented 💯.

## Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lameck1** [comment]
- [OPTIONAL] These two buttons need not be within a form when we consider structuring our page semantically. They can just be two separate buttons that lead to some path.

  Did you have any special reasons for wrapping these two in a form?
  
  Forms are for collecting and submitting user data. I would like to recommend using forms only when necessary.
  
  That aside, currently, a user can create a like when the like button is clicked. A user can also create a comment. But, the comment counts and likes counts are not getting updated in the UI.
  
  Please look into this. Especially, for the like, a user will only realize a like has been created if the count can update.
  
  Screenshot 2022-06-09 210056

  
  ### There is evidence that we can create likes 👇🏼.
  
  image

**Comment by Behnam1369** [comment]
@Lameck1 Thank you for your constructive feedback. I implemented your suggestion about updating likes_counter and comments_counter after every change. 
The `like/unlike` button is submitting a request and it is not redirecting the user to a different page. This is why I wrapped this button inside a form. About the `Add comment` button you are right. I don't need to include it in a form and this is just for easier and consistent styling.

**Pull request merged**